### PR TITLE
KUBE-181 Adding ankh inspect functionality

### PR DESCRIPTION
**Pull request by mmoresco**
Adds commands `ankh inspect values`, `ankh inspect chart`, and `ankh inspect templates`. `ankh inspect values` has the optional flag of `--use-context` to filter the output based on the current ankh context - by default, it will output values for all resource profiles/environments.

### CODE REVIEW

File: src/ankh/helm/helm.go [hunk 1]
```diff
@@ -249,3 +273,155 @@ func Template(ctx *ankh.ExecutionContext, ankhFile ankh.AnkhFile) (string, error
 
 	return finalOutput, nil
 }
+
+func getChartFileContent(ctx *ankh.ExecutionContext, path string, useContext bool, key string) ([]byte, error) {
```
Comment (esmet) [inline src/ankh/helm/helm.go]: this seems like a good opportunity to make a new file `inspect.go`

File: src/ankh/cmd/ankh/main.go [hunk 2]
```diff
@@ -182,6 +182,50 @@ func main() {
 		}
 	})
 
+	app.Command("inspect", "Inspect charts in ankh.yaml and display information.", func(cmd *cli.Cmd) {
+
+		cmd.Command("values", "For each chart, display contexts of values.yaml, "+
```
Comment (esmet) [inline src/ankh/cmd/ankh/main.go]: contents instead of contexts

File: src/ankh/helm/helm.go [hunk 3]
```diff
@@ -249,3 +273,155 @@ func Template(ctx *ankh.ExecutionContext, ankhFile ankh.AnkhFile) (string, error
...
+			}
+		}
+
+		result = bytes
+	} else {
+		ctx.Logger.Debugf("%s not found", path)
+	}
+
+	if len(bytes) > 0 {
+		result = append([]byte("---\n# Source: "+path+"\n"), bytes...)
```
Comment (jondlm) [inline src/ankh/helm/helm.go]: Maybe also include the `Chart: ` bit before this line as well

**Comment by jondlm** [comment]
My comment earlier about a bug was incorrect. I had a messed up config file.

File: src/ankh/helm/helm.go (hunk 1 above)
Comment (mmoresco) [reply src/ankh/helm/helm.go]: Done, added under the `helm` package - lmk if there's a more appropriate location

File: src/ankh/cmd/ankh/main.go (hunk 2 above)
Comment (mmoresco) [reply src/ankh/cmd/ankh/main.go]: whoops, fixed

File: src/ankh/cmd/ankh/main.go [hunk 4]
```diff
@@ -256,9 +300,30 @@ func main() {
...
+	check(err)
+
+	if len(ankhFile.Charts) > 0 {
+		ctx.Logger.Debug("Inspecting charts")
+		for _, chart := range ankhFile.Charts {
+			output, err := cb(ctx, chart, ankhFile)
+			check(err)
+			result += output
+		}
+	}
```
[the remark this answers is not in the document]
Comment (mmoresco) [reply src/ankh/cmd/ankh/main.go]: Done, also added that message to the template command

File: src/ankh/helm/helm.go [hunk 5]
```diff
@@ -249,3 +273,155 @@ func Template(ctx *ankh.ExecutionContext, ankhFile ankh.AnkhFile) (string, error
...
+	}
+
+	return result, nil
+}
+
+func InspectValues(ctx *ankh.ExecutionContext, chart ankh.Chart, ankhFile ankh.AnkhFile) (string, error) {
+	var result string
+
+	ctx.Logger.Debug("Inspecting values for chart %s", chart.Name)
```
[the remark this answers is not in the document]
Comment (mmoresco) [reply src/ankh/helm/helm.go]: Oops, fixed!

**Pull request merged**